library/choose_disk.py

```diff
@@ -0,0 +1,453 @@
+    '''
+    Return a list of keys where ceph disks are reported first
+    while keeping the list sorted
+    '''
+    ceph_disks = []
+    non_ceph_disks = []
+    for physical_disk in sorted(physical_disks):
+        if "ceph_prepared" in physical_disks[physical_disk]:
+            ceph_disks.append(physical_disk)
+        else:
+            non_ceph_disks.append(physical_disk)
+
+    return ceph_disks + non_ceph_disks
+
+
+def find_match(physical_disks, lookup_disks, module=None):
+    ''' Find a set of matching devices in physical_disks
+    '''
+    matched_devices = {}
+    ignored_devices = []
+
+    logger.info("Looking for matches")
+    # Inspecting every disk we search for
+    for disk in sorted(lookup_disks):
+
+        current_lookup = dict(lookup_disks[disk])
+        infinite = False
+        current_type = ""
+        if "infinite" in current_lookup:
+            infinite = True
+            del current_lookup["infinite"]
+
+        # We cannot keep the disk type as a feature to lookup
+        if "ceph_type" in current_lookup.keys():
+            current_type = current_lookup["ceph_type"]
+            del current_lookup["ceph_type"]
+
+        if len(ignored_devices) == len(physical_disks):
+            logger.info(" Skipping %s as no more free devices to match" % (disk))
+            continue
+
+        logger.info(" Inspecting %s" % disk)
+        # Trying to find a match against all physical disks we have
+        for physical_disk in get_keys_by_ceph_order(physical_disks):
+            # Avoid reusing an already matched physical disk
+            if physical_disk in ignored_devices:
+                continue
+
+            current_physical_disk = physical_disks[physical_disk]
+            match_count = 0
+            # Checking what features are matching
+            for feature in current_lookup:
+                if feature not in current_physical_disk:
+                    continue
+
+                # Default operator is equal
+                operator = "_equal"
+
+                # Assign left and right operands
+                right = current_lookup[feature]
+                left = current_physical_disk[feature]
+
+                # Test if we have anoter operator
+                arguments = _REGEXP.search(right)
+                if arguments:
+                        new_operator = "_" + arguments.group(1)
+                        # Check if the associated function exists
+                        if new_operator in globals():
+                            # and assign operands with the new values
+                            operator = new_operator
+                            right = arguments.group(2)
+                        else:
+                            fatal("Unsupported %s operator in : %s" % (new_operator, right), module)
+
+                # Let's check if (left <operator> right) is True meaning the match is done
+                if globals()[operator](convert_units(left), convert_units(right)):
```

Comment:
I'm really not a fan of how you've created a DSL to handle these operators as I think it unnecessarily obfuscates and complicates this code. However, if we must keep it then we should avoid using ``globals()`` here for all the same reasons we've mentioned before about global scope.

A better approach would be to define a dictionary constant that maps the operator to the appropriate method. Something like:

```
OPERATORS = {
    "=": _equal,
    ">": _gt,
    "<": _lt, 
}

# and usage like

if OPERATORS[operator](convert_units(left), convert_units(right)):

```